### PR TITLE
Add A CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by olijeffers0n**
Adds a contributing file to outline code quality and also a brief guide on how to add a new page.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+4. [Style Guidelines](#style-guidelines)
+5. [Code of Conduct](#code-of-conduct)
+
+## Introduction
+
+The PaperMC Docs repository contains the documentation for the PaperMC project, which is used to provide essential 
+information, instructions, and guidelines to our users, developers, and contributors. Your contributions play a 
+significant role in improving and maintaining the quality of our documentation.
+
+We use Docosaurus to build and deploy the documentation website. The Docs Page is hosted on Vercel, which supports automatic
```
Comment (Machine-Maker) [inline CONTRIBUTING.md:19]:
```suggestion
We use [Docusaurus](https://docusaurus.io/) to build and deploy the documentation website. The documentation site is hosted on Vercel, which supports automatic
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+CI/CD deployments from the main branch. The website is available at [https://docs.papermc.io](https://docs.papermc.io).
+
+Using Docosaurus means that there are guides for how to write the documentation. You can find the Docosaurus documentation at
```
Comment (Machine-Maker) [inline CONTRIBUTING.md:22]:
```suggestion
Using Docusaurus means that there are guides for how to write the documentation. You can find the Docusaurus documentation at
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+5. [Code of Conduct](#code-of-conduct)
+
+## Introduction
+
+The PaperMC Docs repository contains the documentation for the PaperMC project, which is used to provide essential 
+information, instructions, and guidelines to our users, developers, and contributors. Your contributions play a 
+significant role in improving and maintaining the quality of our documentation.
+
+We use Docosaurus to build and deploy the documentation website. The Docs Page is hosted on Vercel, which supports automatic
+CI/CD deployments from the main branch. The website is available at [https://docs.papermc.io](https://docs.papermc.io).
```
Comment (Machine-Maker) [inline CONTRIBUTING.md:20]: Vercel also does deployments for all PR branches, though I'm not too sure how important it is to even mention vercel here, its not really material to contributions except that when you open a PR, vercel deploys that in a separate instance to you and others can view the site.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+Using Docosaurus means that there are guides for how to write the documentation. You can find the Docosaurus documentation at
+[https://docusaurus.io/docs/category/guides](https://docusaurus.io/docs/category/guides). The documentation is written
```
Comment (Machine-Maker) [inline CONTRIBUTING.md:23]: Saying `The documentation` after saying where to find the Docusaurus documentation makes it sound like *Docusaurus'* documentation is written in markdown, when I think you are talking about our documentation. Maybe just make the first occurrence of the word Docusaurus a hyperlink to the docs.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+1. Fork the repository to your GitHub account.
+
+2. Create a new branch from the `main` branch.
+
+3. Make your changes or add new documentation following the style guidelines mentioned below.
+
+4. Commit your changes and push them to your fork.
+
+5. Submit a pull request (PR) to the `main` branch of the Docs repository.
```
Comment (Machine-Maker) [inline CONTRIBUTING.md:65]: Perhaps here is a good place to mention vercel deployments of your PR.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+6. Your PR will be reviewed, and feedback may be provided if necessary.
+
+7. Once your contribution meets the guidelines and requirements, it will be merged into the main repository.
+
+## Style Guidelines
+
+To maintain consistency and readability across the documentation, please adhere to the following style guidelines:
+
+1. **American English**: All documentation should be written in American English. Please use American spellings, grammar, and punctuation.
```
Comment (Machine-Maker) [inline CONTRIBUTING.md:75]: Gotta be a website somewhere that covers the main differences for anyone who doesn't know all the quirks. Could be good to link that here

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 6 above)
Comment (MiniDigger) [reply CONTRIBUTING.md:75]: idk if the site is any good, but they list the Our/or, ize/ise, yse/yze ae-oe/e, ence/ense, ogue/og spellings quite nice: https://www.oxfordinternationalenglish.com/differences-in-british-and-american-spelling/

Comment (olijeffers0n) [reply CONTRIBUTING.md:75]: it’s got the word oxford in it so shrug

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+5. [Code of Conduct](#code-of-conduct)
+
+## Introduction
+
+The PaperMC Docs repository contains the documentation for the PaperMC project, which is used to provide essential 
+information, instructions, and guidelines to our users, developers, and contributors. Your contributions play a 
+significant role in improving and maintaining the quality of our documentation.
+
+We use [Docusaurus](https://docusaurus.io/) to build and deploy the documentation website. 
+Using Docusaurus means that there are guides for how to write the documentation. You can find the Docusaurus documentation
```
Comment (Machine-Maker) [inline CONTRIBUTING.md:20]: This `Using Docusaurus means that there are guides...` bit feels out of place. Perhaps combine that and the next sentence to something like
`Docusarus has extensive documentation detailing what is possible with it.` and then link over `extensive documentation`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+[here](https://docusaurus.io/docs/category/guides).
+
+Our Docs are written in Markdown, which is a simple markup language that is easy to learn. 
+Take a look at some of the existing documentation to get a feel for how it is written.
```
Comment (Machine-Maker) [inline CONTRIBUTING.md:24]: I would say, instead of "our docs", "The bulk of Paper's documentation is written with Markdown"

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

**Comment by olijeffers0n** [comment]
Comments Addressed

**Review by Machine-Maker — APPROVED** [review]

**Review by lynxplay — APPROVED** [review]

**Pull request merged**